### PR TITLE
Issue#50 introduce api test framework

### PR DESCRIPTION
**Pull request by Romanov-Niko**
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: home-api-tests/pom.xml [hunk 1]
```diff
@@ -2,15 +2,313 @@
...
+                            <execution>
+                                <id>stop</id>
+                                <phase>post-integration-test</phase>
+                                <goals>
+                                    <goal>stop</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
```
Comment (kh0ma) [inline home-api-tests/pom.xml:254]: Please move this definition to the `<id>api-tests</id>` profile

File: home-api-tests/pom.xml [hunk 2]
```diff
@@ -2,15 +2,313 @@
...
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 
+    <dependencies>
+        <dependency>
+            <groupId>com.softserveinc.ita.homeproject</groupId>
+            <artifactId>home-data-migration</artifactId>
```
Comment (kh0ma) [inline home-api-tests/pom.xml:282]: I think it is mandatory to add the same for home-application

File: home-api-tests/pom.xml [hunk 3]
```diff
@@ -2,15 +2,313 @@
...
+        <liquibase.container.name>liquibase</liquibase.container.name>
+        <liquibase.internal.port>8000</liquibase.internal.port>
+        <home.application.internal.port>8080</home.application.internal.port>
+        <docker.artifacts.path>${project.build.directory}/docker/artifacts</docker.artifacts.path>
+    </properties>
 
-    <artifactId>home-api-tests</artifactId>
+    <profiles>
+        <profile>
+            <id>docker</id>
```
Comment (kh0ma) [inline home-api-tests/pom.xml]: Let's rename this profile to `api-tests-infrastructure`

File: home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/NewsApiIT.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import com.softserveinc.ita.homeproject.model.UpdateNews;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class NewsApiIT {
+
+    private final NewsApi newsApi;
+
+    {
```
Comment (kh0ma) [inline home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/NewsApiIT.java]: What is it about moving this code block to util class?

File: home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/UserApiIT.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        assertEquals(expected.getEmail(), actual.getEmail());
+    }
+
+    private void assertUser(ReadUser saved, UpdateUser update, ReadUser updated) {
+        assertNotNull(updated);
+        assertNotEquals(saved, updated);
+        assertEquals(update.getFirstName(), updated.getFirstName());
+        assertEquals(update.getLastName(), updated.getLastName());
+    }
+}
```
Comment (kh0ma) [inline home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/UserApiIT.java]: `No newline`

**Review by kh0ma — CHANGES_REQUESTED** [review]

File: home-api-tests/pom.xml (hunk 1 above)
Comment (Romanov-Niko) [reply home-api-tests/pom.xml:254]: Done

File: home-api-tests/pom.xml (hunk 3 above)
Comment (Romanov-Niko) [reply home-api-tests/pom.xml]: Done

File: home-api-tests/pom.xml (hunk 2 above)
Comment (Romanov-Niko) [reply home-api-tests/pom.xml:282]: It makes conversion errors. For example `javax.ws.rs.core.Response cannot be converted to com.softserveinc.ita.homeproject.model.ReadN
ews`

File: home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/NewsApiIT.java (hunk 4 above)
Comment (Romanov-Niko) [reply home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/NewsApiIT.java]: Done

File: home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/UserApiIT.java (hunk 5 above)
Comment (Romanov-Niko) [reply home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/UserApiIT.java]: Done

**Review by Dandria1802 — APPROVED** [review]

**Review by kh0ma — APPROVED** [review]

**Review by alexorlenko — APPROVED** [review]

**Review by PoLytvy — APPROVED** [review]

**Pull request merged**